client/service/src/builder.rs

```diff
@@ -1046,13 +1046,26 @@ ServiceBuilder<
 			);
 			let system = system::System::new(system_info, system_rpc_tx.clone());
 
-			sc_rpc_server::rpc_handler((
-				state::StateApi::to_delegate(state),
-				chain::ChainApi::to_delegate(chain),
-				author::AuthorApi::to_delegate(author),
-				system::SystemApi::to_delegate(system),
-				rpc_extensions.clone(),
-			))
+			match offchain_storage.clone() {
+				Some(storage) => {
+					let offchain = sc_rpc::offchain::Offchain::new(storage);
+					sc_rpc_server::rpc_handler((
```

Comment:
@tomusdrw it would be nice if `rpc_handlers` would support `Option<>`, so we don't have this repetition here.

Reply:
Done: https://github.com/paritytech/jsonrpc/pull/535